src/main.cpp

```diff
@@ -1520,6 +1520,10 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
                 __func__, hash.ToString(), FormatStateMessage(state));
         }
 
+        // Check for non-standard witness in P2WSH after transaction is validated
```

Comment:
Why is this being checked after transaction validation?  I thought that the goal of these segwit policy limits was to mitigate potential DoS risk from unwieldy scripts that we haven't thought of; is there any downside to having this happen before transaction validation?